### PR TITLE
disable yield var

### PR DESCRIPTION
**Pull request by czpmango**
Fixes [https://github.com/vesoft-inc/nebula/issues/2960](https://github.com/vesoft-inc/nebula/issues/2960)
fixes [https://github.com/vesoft-inc/nebula/issues/3211](https://github.com/vesoft-inc/nebula/issues/3211)

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
How about other clauses? For example variable in `where`?

**Comment by czpmango** [comment]
> How about other clauses? For example variable in `where`?

The other clause behavior remains as before.

File: tests/tck/features/bugfix/VariableExpression.feature [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
```
Comment (Aiee) [inline tests/tck/features/bugfix/VariableExpression.feature:4]: Remove the CC license

Comment (czpmango) [reply tests/tck/features/bugfix/VariableExpression.feature:4]: ACK.

**Review by CPWstatic — APPROVED** [review]

**Review by jievince — APPROVED** [review]

File: tests/tck/features/bugfix/VariableExpression.feature (hunk 1 above)
Comment (yixinglu) [reply tests/tck/features/bugfix/VariableExpression.feature:4]: It seems you haven't fix the license declaration.

**Review by darionyaphet — DISMISSED** [review]

File: src/parser/parser.yy [hunk 2]
```diff
@@ -1448,9 +1448,26 @@ yield_column
...
     | expression KW_AS name_label {
+        auto vars = graph::ExpressionUtils::collectAll($1, {Expression::Kind::kVar});
+        if (!vars.empty()) {
+            for (auto* var : vars) {
+                if (!static_cast<const VariableExpression*>(var)->isInner()) {
+                    delete $3;
+                    throw nebula::GraphParser::syntax_error(@1, "Direct output of variable is prohibited");
+                }
+            }
+        }
```
Comment (yixinglu) [inline src/parser/parser.yy:1470]: Maybe Given a util function in ExprUtils is better, signature likes:

```c++
bool checkInnerVarExprIfExist(expr);

// usage
if (ExprUtils::checkInnerVarExprIfExit($1)) {
  delete $1;
  throw Exception("...");
}
```

**Review by yixinglu — APPROVED** [review]
LGTM

**Review by jievince — APPROVED** [review]

**Pull request merged**